text/3424-cargo-script.md

```diff
@@ -0,0 +1,593 @@
+/ blogs when teaching.
+
+**Interoperability:**
+
+One angle to look at proposals is if there is a single obvious
+solution.  While this isn't the case for single-file packages, there is enough of
+a subset of one. By standardizing that subset, we allow greater
+interoperability between solutions (e.g.
+[playground could gain support](https://users.rust-lang.org/t/call-for-contributors-to-the-rust-playground-for-upcoming-features/87110/14?u=epage)
+).  This would make it easier to collaborate..
+
+**Prototyping:**
+
+Currently to prototype or try experiment with APIs or the language, you need to either
+- Use the playground
+  - Can't access local resources
+  - Limited in the crates supported
+  - *Note:* there are alternatives to the playground that might have fewer
+    restrictions but are either less well known or have additional
+    complexities.
+- Find a place to do `cargo new`, edit `Cargo.toml` and `main.rs` as necessary, and `cargo run` it, then delete it
+  - This is a lot of extra steps, increasing the friction to trying things out
+  - This will fail if you create in a place that `cargo` will think it should be a workspace member
+
+By having a single-file package,
+- It is easier to setup and tear down these experiments, making it more likely to happen
+- All crates will be available
+- Local resources are available
+
+**One-Off Utilities:**
+
+It is fairly trivial to create a bunch of single-file bash or python scripts
+into a directory and add it to the path.  Compare this to rust where
+- `cargo new` each of the "scripts" into individual directories
+- Create wrappers for each so you can access it in your path, passing `--manifest-path` to `cargo run`
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+As an eRFC, this is meant to convey what we are looking to
+accomplish.  Many of the details may change before
+stablization.
+
+### Creating a New Package
+
+*(Adapted from [the cargo book](https://doc.rust-lang.org/cargo/guide/creating-a-new-project.html))*
+
+To start a new [package][def-package] with Cargo, create a file named `hello_world.rs`:
+```rust
+#!/usr/bin/env cargo
+
+fn main() {
+    println!("Hello, world!");
+}
+```
+
+Let's run it
+```console
+$ chmod +x hello_world.rs
+$ ./hello_world.rs
+Hello, world!
+```
+
+### Dependencies
+
+*(Adapted from [the cargo book](https://doc.rust-lang.org/cargo/guide/dependencies.html))*
+
+[crates.io] is the Rust community's central [*package registry*][def-package-registry]
+that serves as a location to discover and download
+[packages][def-package]. `cargo` is configured to use it by default to find
+requested packages.
+
+#### Adding a dependency
+
+To depend on a library hosted on [crates.io], you modify `hello_world.rs`:
+```rust
+#!/usr/bin/env cargo
+
+//! ```cargo
+//! [dependencies]
+//! time = "0.1.12"
+//! ```
+
+fn main() {
+    println!("Hello, world!");
```

Comment:
Nit, but the code here in this example should actually use the added dependency.

Reply:
(another concerned raised is how stable the version numbers are)